### PR TITLE
view_interface empty() for SizedRange

### PR DESCRIPTION
**Pull request by tower120**
Implement https://github.com/ericniebler/range-v3/issues/1147

### CODE REVIEW

File: include/range/v3/view_interface.hpp [hunk 1]
```diff
@@ -90,6 +90,26 @@ namespace ranges
             {
                 return Cardinality == 0;
             }
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(
+                    Cardinality < 0 && Cardinality != infinite &&
+                    !ForwardRange<D>() &&
+                    SizedRange<D>())>
+            RANGES_CXX14_CONSTEXPR bool empty()
```
Comment (johelegp) [inline include/range/v3/view_interface.hpp]: This qualifies for C++11 `constexpr`.

File: include/range/v3/view_interface.hpp [hunk 2]
```diff
@@ -90,6 +90,26 @@ namespace ranges
...
+                noexcept(noexcept(bool(ranges::size(std::declval<D &>()) == 0)))
+            {
+                return ranges::size(derived()) == 0;
+            }
+            template<typename D = Derived,
+                CONCEPT_REQUIRES_(
+                    Cardinality < 0 && Cardinality != infinite &&
+                    !ForwardRange<D const>() &&
+                    SizedRange<D const>())>
+            RANGES_CXX14_CONSTEXPR bool empty() const
```
Comment (johelegp) [inline include/range/v3/view_interface.hpp]: And this too.

File: include/range/v3/view_interface.hpp (hunk 1 above)
Comment (johelegp) [reply include/range/v3/view_interface.hpp]: Actually, not this one due to the lack of `const`.

Comment (tower120) [reply include/range/v3/view_interface.hpp]: ??

Comment (johelegp) [reply include/range/v3/view_interface.hpp]: The other `empty` overload you added can be `constexpr` in C++11, so you should replace `RANGES_CXX14_CONSTEXPR` for `constexpr`. Not this one, though.

**Pull request merged**